app/upgrades/v17/upgrades.go

```diff
@@ -0,0 +1,327 @@
+//
+// NOTE: If a parameter is added, the old params cannot be unmarshalled
+// to the new schema. To get around this, we have to set each parameter explicitly
+// Considering all mainnet stakeibc params are set to the default, we can just use that
+func MigrateStakeibcParams(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	params := stakeibctypes.DefaultParams()
+	k.SetParams(ctx, params)
+}
+
+// Migrates the host zones to the new structure which supports community pool liquid staking
+// We don't have to perform a true migration here since only new fields were added
+// (in other words, we can deserialize the old host zone structs into the new types)
+// This will also register the relevant community pool ICA addresses
+func RegisterCommunityPoolAddresses(ctx sdk.Context, k stakeibckeeper.Keeper) error {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		chainId := hostZone.ChainId
+
+		// Create and store a new community pool stake and redeem module address
+		stakeHoldingAddress := stakeibctypes.NewHostZoneModuleAddress(
+			chainId,
+			stakeibckeeper.CommunityPoolStakeHoldingAddressKey,
+		)
+		redeemHoldingAddress := stakeibctypes.NewHostZoneModuleAddress(
+			chainId,
+			stakeibckeeper.CommunityPoolRedeemHoldingAddressKey,
+		)
+
+		if err := utils.CreateModuleAccount(ctx, k.AccountKeeper, stakeHoldingAddress); err != nil {
+			return errorsmod.Wrapf(err, "unable to create community pool stake account for host zone %s", chainId)
+		}
+		if err := utils.CreateModuleAccount(ctx, k.AccountKeeper, redeemHoldingAddress); err != nil {
+			return errorsmod.Wrapf(err, "unable to create community pool redeem account for host zone %s", chainId)
+		}
+
+		hostZone.CommunityPoolStakeHoldingAddress = stakeHoldingAddress.String()
+		hostZone.CommunityPoolRedeemHoldingAddress = redeemHoldingAddress.String()
+
+		k.SetHostZone(ctx, hostZone)
+
+		// Register the deposit and return ICA addresses
+		// (these will get set in the OnChanAck callback)
+		// create community pool deposit account
+		connectionId := hostZone.ConnectionId
+		connectionEnd, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, connectionId)
+		if !found {
+			return errorsmod.Wrapf(connectiontypes.ErrConnectionNotFound, "connection %s not found", connectionId)
+		}
+		counterpartyConnectionId := connectionEnd.Counterparty.ConnectionId
+
+		appVersion := string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
+			Version:                icatypes.Version,
+			ControllerConnectionId: connectionId,
+			HostConnectionId:       counterpartyConnectionId,
+			Encoding:               icatypes.EncodingProtobuf,
+			TxType:                 icatypes.TxTypeSDKMultiMsg,
+		}))
+
+		depositAccount := stakeibctypes.FormatHostZoneICAOwner(chainId, stakeibctypes.ICAAccountType_COMMUNITY_POOL_DEPOSIT)
+		if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, depositAccount, appVersion); err != nil {
```

Comment:
Could a registration fail / is there any way for us to recover if one does?

Reply:
Did you mean the submission itself or the full account creation?

I think the submission can only fail if the connection is closed. I think localstride should give us enough coverage here

I'm not sure what could cause the channel handshake and full account creation to fail, but I don't think it's any different than when we normally register a host

Reply:
> I'm not sure what could cause the channel handshake and full account creation to fail, but I don't think it's any different than when we normally register a host

Makes sense. Can we use the standard `RestoreInterchainAccount` to restore closed channels / did you and @ethan-stride test that as part of the DYDX flow?

Reply:
we modified the original tx so that it works with both host zone and trade route channels, and yeah we tested it

Reply:
Sounds good!